File: 1915-check-if-one-string-swap-can-make-strings-equal/1915-check-if-one-string-swap-can-make-strings-equal.py
class Solution:
    def areAlmostEqual(self, s1: str, s2: str) -> bool:
        index=[]
        for i in range(len(s1)):
            if(s1[i]!=s2[i]):
                index.append(i)

                if len(index)>2:
                    return False
        if len(index)==2:
            i,j=index
            return s1[i]==s2[j] and s1[j]==s2[i]
        return len(index)==0

File: 1915-check-if-one-string-swap-can-make-strings-equal/1915-check-if-one-string-swap-can-make-strings-equal.java
class Solution {
    public boolean areAlmostEqual(String s1, String s2) {
        int p1=-1,p2=-1;
        int count=0;
        for(int i=0;i<s1.length();i++){
            if(s1.charAt(i)!=s2.charAt(i)){
                count++;
                if(count>2) return false;
                if(p1==-1) p1=i;
                else p2=i;
            }
        }
        if(count==0) return true;
        if(count!=2) return false;
        return s1.charAt(p1)==s2.charAt(p2) && s1.charAt(p2)==s2.charAt(p1);
    }
}